sympy/ntheory/qs.py

```diff
@@ -121,7 +122,7 @@ def _initialize_first_polynomial(N, M, factor_base, idx_1000, idx_5000, seed=Non
     """
     if seed is not None:
         rgen.seed(seed)
-    approx_val = sqrt(2*N) / M
+    approx_val = sqrt(Decimal(2*N)) / M
```

Comment:
It would be better to use `isqrt` if `N` is always a positive integer

Reply:
You are right. Using `isqrt` we avoid converting the argument to `float` type. Moreover, `isqrt` returns `int` and removes the limit on the size of result (int in Python3 is equivalent to long in Python2, and there is no limit on the maximum value. It is possible to handle as large value as memory is available.)